### ISSUE TITLE
OkHttpCallUtil.java Error when exiting the app using BackAndroid.exitapp()

### ISSUE THREAD

**Issue by avishayil** [facebook/react-native#8009]
This is the crash log:
http://crashes.to/s/e247dbd6fc2
```java
Fatal Exception: java.lang.RuntimeException: Unable to destroy activity {com.zuznow.geektime/com.zuznow.geektime.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5066)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5089)
       at android.app.ActivityThread.access$1700(ActivityThread.java:222)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1854)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
       at com.facebook.react.common.network.OkHttpCallUtil.cancelTag(OkHttpCallUtil.java:30)
       at com.facebook.react.modules.network.NetworkingModule.onCatalystInstanceDestroy(NetworkingModule.java:141)
       at com.facebook.react.bridge.NativeModuleRegistry.notifyCatalystInstanceDestroy(NativeModuleRegistry.java:98)
       at com.facebook.react.bridge.CatalystInstanceImpl.destroy(CatalystInstanceImpl.java:218)
       at com.facebook.react.bridge.ReactContext.destroy(ReactContext.java:193)
       at com.facebook.react.ReactInstanceManagerImpl.destroy(ReactInstanceManagerImpl.java:510)
       at com.facebook.react.ReactActivity.onDestroy(ReactActivity.java:177)
       at android.app.Activity.performDestroy(Activity.java:7102)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1170)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5044)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5089)
       at android.app.ActivityThread.access$1700(ActivityThread.java:222)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1854)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

Happens sometimes.
Hope you guys can help / fix this.

Thanks!

**Comment by satya164** [comment]
I edited your post to inline the crash log. Hope you don't mind.

**Comment by charpeni** [comment]
@facebook-github-bot label Android

**Comment by suxycn** [comment]
+1

**Comment by alwx** [comment]
Seems like it is fixed in the current master branch, because `NetworkingModule.onCatalystInstanceDestroy` method doesn't contain any calls to `OkHttpCallUtil.cancelTag` anymore.

**Comment by ajwhite** [comment]
@alwx it still does 

`NetworkingModule.onCatatalystInstanceDestroy` calls [cancelAllRequests](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L143),

which [iterates over all pending requests](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L337-L342) and calls `cancelRequest`, which [peforms `OkHttpCallUtil.cancelTag`](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L366-L375) still.

**Comment by ahanriat** [comment]
+1

**Comment by ajwhite** [comment]
This has been reported fixed in `0.28` https://github.com/facebook/react-native/issues/7755
